### PR TITLE
feat(empty-stacktraces): Flag-protect issue search for the empty_stacktrace.js_console tag

### PR DESCRIPTION
**Pull request by snigdhas**
Disallow searching for issues with the empty_stacktrace.js_console tag if the `organizations:javascript-console-error-tag` is not set.

See https://github.com/getsentry/sentry/pull/39335 and WOR-2187.

### CODE REVIEW

**Review by scttcper — APPROVED** [review]

File: src/sentry/api/endpoints/organization_tags.py [hunk 1]
```diff
@@ -29,6 +29,13 @@ def get(self, request: Request, organization) -> Response:
                     # Defaults to True, because the frontend caches these tags globally
                     include_transactions=request.GET.get("include_transactions", "1") == "1",
                 )
+                if not features.has(
+                    "organizations:javascript-console-error-tag",
+                    organization,
+                    actor=None,
+                ):
+                    results = [tag for tag in results if tag != "empty_stacktrace.js_console"]
```
Comment (armenzg) [inline src/sentry/api/endpoints/organization_tags.py:37]: Is this so the tag does *not* show up on the list of tags to pick from?

File: src/sentry/api/issue_search.py [hunk 2]
```diff
@@ -142,7 +142,15 @@ def convert_query_values(search_filters, projects, user, environments):
     :return: New collection of `SearchFilters`, which may have converted values.
     """
 
-    def convert_search_filter(search_filter):
+    def convert_search_filter(search_filter, organization):
+        if search_filter.key.name == "empty_stacktrace.js_console":
```
Comment (armenzg) [inline src/sentry/api/issue_search.py:146]: The idea here is that you're preventing anyone who is not allowed to try the tag via the UI; correct?

File: tests/sentry/api/test_issue_search.py [hunk 3]
```diff
@@ -158,6 +158,22 @@ def test_parens_in_query(self):
...
 
 
+class ConvertJavaScriptConsoleTagTest(TestCase):
+    def test_valid(self):
+        filters = [SearchFilter(SearchKey("empty_stacktrace.js_console"), "=", SearchValue(True))]
+        with self.feature("organizations:javascript-console-error-tag"):
+            result = convert_query_values(filters, [self.project], self.user, None)
+            assert result[0].value.raw_value is True
+
+    def test_invalid(self):
```
Comment (armenzg) [inline tests/sentry/api/test_issue_search.py:168]: Good job covering both cases.

**Review by armenzg — APPROVED** [review]

File: src/sentry/api/issue_search.py (hunk 2 above)
Comment (snigdhas) [reply src/sentry/api/issue_search.py:146]: yup!

File: src/sentry/api/endpoints/organization_tags.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/api/endpoints/organization_tags.py:37]: yes, we don't want users from other orgs to see the tag.

**Pull request merged**